packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1522,10 +1503,10 @@ function commitDeletion(
     // Detach refs and call componentWillUnmount() on the whole subtree.
     commitNestedUnmounts(finishedRoot, current, renderPriorityLevel);
   }
+  current.return = null;
```

Comment:
These are needed to make the failing test I mentioned in the comments of this PR pass, as per @sebmarkbage's response. If we could avoid doing this, it would be better.